### DISCUSSION TITLE
Security/Firewall Setup and High Availability

### DISCUSSION THREAD

**Discussion by acaz** [gravitational/teleport#7705] [Q&A]
Hi 

First off all thanks to your team. Teleport is a great product.
I have setup a teleport cluster to provide ssh , databases and applications access   :  1 auth server and 1 proxy server and some nodes.

I read lot of teleport's documentation but could you please clarify some points :

**First question**
My firewall configuration :
- the auth server expose tcp port 3025 to any internet ip
- the proxy server expose tcp port 3080 (webui) , 3023 (proxy), 3024 (reverse tunnel)  and 3036 ( mysql service ) to any internet ip
- teleport node expose tcp port  3022  to proxy public ip 

Is it the right way ?
Should the firewall's rules being more restrictiv ? 

**Second question**
We use dynamodb as backend storage. 
In this case, could we use multiple auth server "simultaneously" ( adding an auth server,  running in a private network  for example )  ?

Advices are welcome.

Regards 
Arnaud

**Comment by stevenGravy** [comment]
hi @acaz , the auth server '3025` is only required to be available to the proxy and any node or other teleport service connecting to it.  Would not recommend having it public.
The `3024` reverse tunnel port on the proxy is only required to be available for nodes connecting.

**Reply by acaz** [reply]
hi @stevenGravy  
Thanks for those information.

for auth server :  
i closed tcp port 3025 to public access , allow only proxy and node ( it will be not easy for node, lot of public ip to allow  )

for proxy:
i let only  open web ui (3080) to public 
i will restrict the reverse tunnel port 3024  to node only or allow direct connexion to node ( 3022 ) , is that right ?  is it better to setup a second  proxy server ( trusted cluster )  ? 

Should i restrict  access to 
3023 port to tsh client ip 
3036 port ( db proxy ) to tsh client ip 
or could i let them open to any ip ? 

It's a bunch of questions, but we want use teleport in a secure way. 
 
Regards
Arnaud
